nemo/collections/tts/inference/pipeline.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+import numpy as np
+import torch
+
+from nemo.utils.decorators import experimental
+
+
+class TextProcessor(ABC):
+    @abstractmethod
+    def preprocess_text(self, text: str) -> str:
+        """
+        Pre-normalization text processing.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            pre-processed text string.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def normalize_text(self, text: str) -> str:
+        """
+        Text normalization.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            normalized text string.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def postprocess_text(self, text: str) -> str:
+        """
+        Post-normalization text processing.
+
+        Args:
+            text: input text string.
+
+        Returns:
+            post-processed text string.
+        """
+        raise NotImplementedError()
+
+
+class GraphemeToPhoneme(ABC):
+    @abstractmethod
+    def convert_graphemes_to_phonemes(self, text: str) -> str:
+        """
+        Convert string containing graphemes/characters to phonemes (eg. IPA or ARPABET).
+          Out of vocabulary words are left as graphemes.
+
+        Args:
+            text: post-processed text string.
+
+        Returns:
+            phoneme string
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def convert_graphemes_to_phonemes_mixed(self, text: str, phone_prob: float) -> str:
```

Comment:
Why have both this and the previous `convert_graphemes_to_phonemes()` if setting `phone_prob` will cover both cases?

Reply:
Gotcha, sounds good.